src/nnue/nnue_accumulator.cpp

```diff
@@ -374,28 +397,28 @@ void update_accumulator_refresh_cache(const FeatureTransformer<Dimensions>& feat
     auto&                 entry = cache[ksq][Perspective];
     FeatureSet::IndexList removed, added;
 
-    for (Color c : {WHITE, BLACK})
+    Piece new_pieces[SQUARE_NB];
+    for (Square sq = SQUARE_ZERO; sq < SQUARE_NB; ++sq)
```

Comment:
If we add an accessor method to return a const pointer to the ```Position::board[SQUARE_NB]``` array you can then pass it to ```get_changed_pieces()``` directly without needing to temporarily copy to the ```new_pieces[]``` array.